lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -0,0 +1,273 @@
+
+                                   // Note: The number of fractional digits specified for a currency is not locale-dependent.
+        icu::NumberFormat *nf = icu::NumberFormat::createCurrencyInstance(error); // using default locale
+        if (U_FAILURE(error))
+        {
+#ifdef INTL_ICU_DEBUG
+            if (error == UErrorCode::U_MISSING_RESOURCE_ERROR)
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > icu::NumberFormat::createCurrencyInstance(error) > U_MISSING_RESOURCE_ERROR (%d)\n"), error);
+            }
+            else
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > icu::NumberFormat::createCurrencyInstance(error) > UErrorCode (%d)\n"), error);
+            }
+#endif
+            goto LReturn;
+        }
+
+        nf->setCurrency(uCurrencyCode, error);
+        if (U_FAILURE(error))
+        {
+#ifdef INTL_ICU_DEBUG
+            if (error == UErrorCode::U_MISSING_RESOURCE_ERROR)
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > nf->setCurrency(uCurrencyCode (%s), error) > U_MISSING_RESOURCE_ERROR (%d)\n"), currencyCode, error);
+            }
+            else
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > nf->setCurrency(uCurrencyCode (%s), error) > UErrorCode (%d)\n"), currencyCode, error);
+            }
+#endif
+            goto LReturn;
+        }
+
+        minFracDigits = nf->getMinimumFractionDigits();
+
+#ifdef INTL_ICU_DEBUG
+        Output::Print(_u("EntryIntl_CurrencyDigits > nf->getMinimumFractionDigits() successful > returned (%d)\n"), minFracDigits);
+#endif
+
+    LReturn:
+        // Since something failed, return "reasonable" default of 2 fractional digits (obviously won't be the case for some currencies like JPY).
+        // REVIEW (doilij): What does the spec say to do if a currency is not supported?
+
+#ifdef INTL_ICU_DEBUG
+        Output::Print(_u("EntryIntl_CurrencyDigits > returning (%d)\n"), minFracDigits);
+#endif
+
+        return minFracDigits;
+    }
+
+    IPlatformAgnosticResource *CreateNumberFormat()
+    {
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+        // REVIEW (doilij): is allocating with `new` here okay?
+        icu::NumberFormat *nf = icu::NumberFormat::createInstance(error);
+        IPlatformAgnosticResource *obj = new PlatformAgnosticIntlObject<icu::NumberFormat>(nf);
+        AssertMsg(U_SUCCESS(error), "Creating icu::NumberFormat failed");
+        return obj;
+    }
+
+    template const char16 *FormatNumber<>(IPlatformAgnosticResource *formatter, int32_t val);
+    template const char16 *FormatNumber<>(IPlatformAgnosticResource *formatter, double val);
+
+    template <typename T>
+    const char16 *FormatNumber(IPlatformAgnosticResource *formatter, T val)
+    {
+        icu::UnicodeString result;
+
+        auto *formatterHolder = (PlatformAgnosticIntlObject<icu::NumberFormat> *)formatter;
```

Comment:
It seems like the codebase generally favors reinterpret_cast?

Reply:
Not so much that it favors (it's also hard to grep for C-style casts) but it is better practice. I'll update.

Reply:
Done